api/wasm.go

```diff
@@ -27,13 +27,19 @@ import (
 //	result := api.DecodeF64(result[0])
 //
 // Note: This is a type alias as it is easier to encode and decode in the binary format.
+// Note: Integer types are encoded as signed numbers, which affects their max value.
+// See https://www.w3.org/TR/wasm-core-1/#value-types%E2%91%A2
 // See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#binary-valtype
 type ValueType = byte
 
 const (
+	// ValueTypeI32 is a 32-bit integer with max value math.MaxInt32.
```

Comment:
Oh I think this comes with confusion.. as what the spec is saying is that its *binary encoding* is signed leb128 (therefore the binary representation of i32 const or whatever has limit on math.MaxInt32), but it doesn't mean the value itself has limit on math.MaxInt32. For example, host function `func (i32, it32)` needs to handle (2**32).

Reply:
For globals, yes, the value must be encoded in text or binary format which means it has the limit on math.MaxtInt32 .

Reply:
gotcha.. I will move this to the global instead!

Reply:
For reference on what `i32` means:
```
The types i32 and i64 classify 32 and 64 bit integers, respectively. Integers are not inherently signed or unsigned, their interpretation is determined by individual operations.
```

Reply:
PTAL!